### PR TITLE
ARROW-17013: [CI][C++] Fix arrow build for Ubuntu CPP 22.04

### PR DESCRIPTION
**Pull request by raulcd**

### CODE REVIEW

**Comment by raulcd** [comment]
@github-actions crossbow submit test-*-cpp

**Comment by raulcd** [comment]
The two R job failures seems to be a known issue: https://issues.apache.org/jira/browse/ARROW-16973 not related with the PR
The Appveyor failure is also unrelated (happening in master right now): https://issues.apache.org/jira/browse/ARROW-17014

**Comment by raulcd** [comment]
@kou @lidavidm this was making builds fail on our nightly for ubuntu-22.04-cpp. It was not failing for ubuntu 18.04 nor 20.4 because flight and flight sql is not enabled there.
I am not an expert on CMake but all builds are working with this being removed so I am not entirely sure whether it was necessary  but this is fixing the build failure. Let me know what you think about this.

File: cpp/src/arrow/flight/sql/CMakeLists.txt [hunk 1]
```diff
@@ -117,10 +117,4 @@ if(ARROW_BUILD_TESTS OR ARROW_BUILD_EXAMPLES)
   add_executable(flight-sql-test-app test_app_cli.cc)
   target_link_libraries(flight-sql-test-app PRIVATE ${ARROW_FLIGHT_SQL_TEST_LINK_LIBS}
                                                     ${GFLAGS_LIBRARIES})
-
-  if(ARROW_FLIGHT_TEST_LINKAGE STREQUAL "static" AND ARROW_BUILD_STATIC)
-    target_compile_definitions(flight_sql_test
```
Comment (lidavidm) [inline cpp/src/arrow/flight/sql/CMakeLists.txt:122]: Oops, my bad. We still need this block, though, I believe. Just replace `flight_sql_test` with `arrow-flight-sql-test`.

Comment (raulcd) [reply cpp/src/arrow/flight/sql/CMakeLists.txt:122]: it does fail on that case:
```
CMake Error at src/arrow/flight/sql/CMakeLists.txt:122 (target_compile_definitions):
  target_compile_definitions called with invalid arguments
```
with:
```
  if(ARROW_FLIGHT_TEST_LINKAGE STREQUAL "static" AND ARROW_BUILD_STATIC)
    target_compile_definitions(arrow-flight-sql-test
                               flight-sql-test-server flight-sql-test-app
                               PUBLIC ARROW_FLIGHT_STATIC ARROW_FLIGHT_SQL_STATIC)
  endif()
```

Comment (lidavidm) [reply cpp/src/arrow/flight/sql/CMakeLists.txt:122]:
```diff
   if(ARROW_FLIGHT_TEST_LINKAGE STREQUAL "static" AND ARROW_BUILD_STATIC)
-    target_compile_definitions(flight_sql_test
-                               flight-sql-test-server flight-sql-test-app
-                               PUBLIC ARROW_FLIGHT_STATIC ARROW_FLIGHT_SQL_STATIC)
+    foreach(TEST_TARGET arrow-flight-sql-test flight-sql-test-server flight-sql-test-app)
+      target_compile_definitions(${TEST_TARGET}
+                                PUBLIC ARROW_FLIGHT_STATIC ARROW_FLIGHT_SQL_STATIC)
+    endforeach()
   endif()
```

**Comment by raulcd** [comment]
@github-actions crossbow submit test-ubuntu-22.04-cpp

**Review by lidavidm — APPROVED** [review]

**Review by kou — APPROVED** [review]
+1

**Pull request merged**